source/engine/graphics/system/LightSystem.h

```diff
@@ -0,0 +1,107 @@
+#pragma once
+
+#include "ecs/Transform.h"
+#include "ecs/View.h"
+#include "graphics/GraphicsConstants.h"
+#include "graphics/shader_resource/ShaderResourceBus.h"
+#include "graphics/shader_resource/PPImageArrayBufferHandle.h"
+#include "graphics/shader_resource/StorageBufferHandle.h"
+#include "graphics/PointLight.h"
+#include "graphics/SpotLight.h"
+#include "utility/Signal.h"
+
+#include "DirectXMath.h"
+
+#include <vector>
+
+namespace nc::graphics
+{
+/** @brief Device-mapped properties of a PointLight. */
+struct PointLightData
+{
+    PointLightData() = default;
```

Comment:
Don't know if it was like this before, but a default constructor without default initializers on all primitives is unpredictable. You could probably just add a default for `radius` and remove both constructors since one is default and the other takes every member.

Reply:
Oh I see, its because of the `isInitialized` member. Still, if the default is here, `radius` needs an initializer.